pkg/reconcile/reconcilers.go

```diff
@@ -178,6 +185,59 @@ func (r *defaultReconciler) ReconcileSchema(ctx context.Context, req ReaperReque
 	}
 }
 
+func (r *defaultReconciler) checkForCassandraDatacenterReadiness(ctx context.Context, req ReaperRequest) error {
+	reaper := req.Reaper
+	cassandra := *reaper.Spec.ServerConfig.CassandraBackend
+	cassdc := &cassdcapi.CassandraDatacenter{}
+
+	fetchStatus := func() error {
+		cassdcKey := types.NamespacedName{Namespace: reaper.Namespace, Name: cassandra.DatacenterName()}
+		err := r.Client.Get(ctx, cassdcKey, cassdc)
+		if runtime.IsNotRegisteredError(err) {
+			return nil
+		}
+		if err != nil {
+			req.Logger.Error(err, "failed to fetch CassandraDatacenter")
+			return err
+		}
+		return nil
+	}
+
+	maxWaitTime := time.Now().Add(30 * time.Second)
+
+	for {
```

Comment:
I think it is generally best to avoid potentially long blocking operations during reconciliation. You want a workflow along these lines:

* Fetch the CassandraDatacenter
* If fetch fails
    * Log the error
    * Requeue the request
* Check the status
* If the CassandraDatacenter is not ready
    * Log a message
    * Request the request
* If the CassandraDatacenter is ready
    * Log a message
    * Continue with reconciliation 

And if we really wanted to get fancy, we could calculate the requeue delay bases on the C* cluster size and the number of up C* nodes. We can  hold off on that for now :)